src/game/client/worldheightmap.cpp

```diff
@@ -29,21 +45,2358 @@ class GDIFileStream : public InputStream
+                            }
+                        }
+                    }
+                } else {
+                    this->m_textureClasses[j].position_in_texture.x = 0;
+                    this->m_textureClasses[j].position_in_texture.y = 0;
+                }
+            }
+        }
+    }
+
+    for (int i = 0; i < m_numBitmapTiles; i++) {
+        if (this->m_edgeTiles[i]) {
+            this->m_edgeTiles[i]->m_tileLocationInTexture.x = 0;
+            this->m_edgeTiles[i]->m_tileLocationInTexture.y = 0;
+        }
+    }
+
+    int i3 = 0;
+
+    for (int i = 0; i < 28; i++) {
+        for (int j = 0; j < 28; j++) {
+            availableGrid[i][j] = 1;
+        }
+    }
+
+    for (int i = 0; i < m_numEdgeTextureClasses; i++) {
+        int width = this->m_edgeTextureClasses[i].width;
+        bool b = 0;
+
+        int j;
+        int l;
+        for (j = 0; j < 28 - width + 1; j++) {
+            for (int k = 0; k < 28 - width + 1; k++) {
+                if (availableGrid[j][k]) {
+                    bool b2 = 1;
+
+                    for (l = 0; l < width && b2; ++l) {
+                        for (int m = 0; m < width && b2; m++) {
+                            if (!availableGrid[j][28 * m + l + k]) {
+                                b2 = 0;
+                            }
+                        }
+                    }
+                    if (b2) {
+                        b = 1;
+                    }
+                    break;
+                }
+            }
+            if (b) {
+                break;
+            }
+        }
+
+        if (b) {
+            int x = 72 * l + 4;
+            int y = 72 * j + 4;
+            this->m_edgeTextureClasses[i].position_in_texture.x = x;
+            this->m_edgeTextureClasses[i].position_in_texture.y = y;
+
+            if (i3 < y + (width << 6) + 4) {
+                i3 = y + (width << 6) + 4;
+            }
+
+            for (int m = 0; m < width; m++) {
+                for (int n = 0; n < width; n++) {
+                    availableGrid[j][28 * n + m + l] = 0;
+                    int i2 = width * n + m + this->m_edgeTextureClasses[i].first_tile;
+
+                    if (this->m_edgeTiles[i2]) {
+                        this->m_edgeTiles[i2]->m_tileLocationInTexture.x = (m << 6) + x;
+                        this->m_edgeTiles[i2]->m_tileLocationInTexture.y = ((width - n - 1) << 6) + y;
+                    }
+                }
+            }
+        } else {
+            this->m_edgeTextureClasses[i].position_in_texture.x = 0;
+            this->m_edgeTextureClasses[i].position_in_texture.y = 0;
+        }
+    }
+
+    if (edge_height) {
+        *edge_height = i3;
+    }
+
+    return i1;
+}
+
+void WorldHeightMap::Get_UV_For_Ndx(int tile_ndx, float *min_u, float *max_u, float *min_v, float *max_v, bool full_tile)
+{
+    if (m_sourceTiles[tile_ndx >> 2]) {

```

Comment:
whole function needs redoing args are `void __thiscall WorldHeightMap__getUVForNdx(WorldHeightMap *this, int tileNdx, float *minU, float *minV, float *maxU, float *maxV, bool fullTile)`

Reply:
fixed.